lib/Page.js

```diff
@@ -136,6 +187,24 @@ Page.prototype.removeFrontMatter = function (includedPage) {
   return $.html();
 };
 
+Page.prototype.insertTemplates = function (pageData) {
+  let modifiedPageData = pageData;
+
+  if (this.frontMatter.footer) {
+    const specifiedFooter = this.frontMatter.footer;
+    const footerFilePath = path.join(this.rootPath, FOOTERS_FOLDER_PATH, specifiedFooter);
+    if (!utils.fileExists(footerFilePath)) {
+      throw new Error(`File ${specifiedFooter} does not exist in ${FOOTERS_FOLDER_PATH}`);
+    }
+
+    const footerData = fs.readFileSync(footerFilePath, 'utf8');
+    const pageUserDefinedVariables = this.userDefinedVariablesMap[path.join(this.rootPath, this.baseUrl)];
+    modifiedPageData += `\n${nunjucks.renderString(footerData, pageUserDefinedVariables)}`;
+  }
+
+  return modifiedPageData;
```

Comment:
- Method was designed initially to house potential future templates to be inserted via specifying in `Front Matter`
- Wouldn't `fs.readFileSync` throw an error if the path is invalid?

Reply:
> - Method was designed initially to house potential future templates to be inserted via specifying in `Front Matter`

Doesn't seem appropriate at this point.

> - Wouldn't `fs.readFileSync` throw an error if the path is invalid?

Yes, with an error no less informative than what is currently thrown.
Especially since we don't intend to recover from it.

See https://nodejs.org/api/fs.html#fs_fs_exists_path_callback:

>> Using `fs.exists()` to check for the existence of a file before calling `fs.open()`, `fs.readFile()` or `fs.writeFile()` is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file does not exist.